x-pack/plugins/log_explorer/public/utils/proxies.ts

```diff
@@ -30,6 +30,30 @@ export const createPropertyGetProxy = <Target extends object, Key extends keyof
     },
   });
 
+const proxySingletonCache = new Map();
```

Comment:
Sounds good to me, I used the Map approach in case we need more proxies, but it's not likely we'll have to create many. I'll switch it to your suggestion, thank you!

Reply:
thanks. the `Map` certainly is more powerful, but comes with risks. While at it, do we want to make sure we keep the inner `() => {}` for `enableStorage` constant too? I forgot that in my proposal.

Reply:
Just fixed, thanks again for the review :)